frontend/src/utils/TokenService.js

```
@@ -18,53 +18,79 @@ const TokenService = {
-      .map(b => b.toString(16).padStart(2, '0'))
-      .join('');
-    const timestamp = Date.now().toString(36);
-    return timestamp + '-' + randomHex;
+    try {
+      const buffer = new Uint8Array(8);
+      window.crypto.getRandomValues(buffer);
+      const randomHex = Array.from(buffer)
+        .map(b => b.toString(16).padStart(2, '0'))
+        .join('');
+      const timestamp = Date.now().toString(36);
+      return timestamp + '-' + randomHex;
```

Biome (1.9.4)

[error] Template literals are preferred over string concatenation.

Unsafe fix: Use a template literal.


(lint/style/useTemplate)